ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/ENRResponsePacketDataTest.java

```diff
@@ -64,17 +100,66 @@ public void readFrom() {
     assertThat(deserialized.getEnr()).isEqualTo(nodeRecord);
   }
 
+  @Test
+  public void writeTo() {
+    final Bytes requestHash = Bytes.fromHexStringLenient("0x1234");
+    final Bytes nodeId =
+        Bytes.fromHexString("a448f24c6d18e575453db13171562b71999873db5b286df957af199ec94617f7");
+    final Bytes privateKey =
+        Bytes.fromHexString("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291");
+
+    NodeRecord nodeRecord =
+        NodeRecordFactory.DEFAULT.createFromValues(
+            UInt64.ONE,
+            new EnrField(EnrField.ID, IdentitySchema.V4),
+            new EnrField(EnrField.IP_V4, Bytes.fromHexString("0x7F000001")),
+            new EnrField(EnrField.UDP, 30303),
+            new EnrField(
+                EnrField.PKEY_SECP256K1, Functions.derivePublicKeyFromPrivate(privateKey)));
+    nodeRecord.sign(privateKey);
+
+    assertThat(nodeRecord.getNodeId()).isEqualTo(nodeId);
+    assertThat(nodeRecord.asEnr())
+        .isEqualTo(
+            "enr:-IS4QHCYrYZbAKWCBRlAy5zzaDZXJBGkcnh4MHcBFZntXNFrdvJjX04jRzjzCBOonrkTfj499SZuOh8R33Ls8RRcy5wBgmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQPKY0yuDUmstAHYpMa2_oxVtw0RW_QAdpzBQA8yWM0xOIN1ZHCCdl8");
+
+    BytesValueRLPOutput out = new BytesValueRLPOutput();
+    final ENRResponsePacketData packet = ENRResponsePacketData.create(requestHash, nodeRecord);
+    packet.writeTo(out);
+    final Bytes encoded = out.encoded();
+
+    final ENRResponsePacketData deserialized = ENRResponsePacketData.readFrom(RLP.input(encoded));
+    assertThat(deserialized.getRequestHash()).isEqualTo(requestHash);
+    assertThat(deserialized.getEnr()).isEqualTo(nodeRecord);
+  }
+
   @Test
   public void readFrom_withExtraFields() {
     final Bytes requestHash = Bytes.fromHexStringLenient("0x1234");
-    final NodeRecord nodeRecord =
-        new NodeRecordFactory(IdentitySchemaInterpreter.V4)
-            .createFromValues(UInt64.ONE, new EnrField(EnrField.ID, IdentitySchema.V4));
+    final Bytes nodeId =
```

Comment:
I think we need a test where reading fails signature verification. I believe it's built into the NodeRecord so we just need to verify isValid fails.  Just take the ENR from this one and change one of the letters near the end.